core/templates/domain/utilities/stopwatch.model.ts

```diff
@@ -22,7 +22,8 @@
 import { LoggerService } from 'services/contextual/logger.service';
 
 export class Stopwatch {
-  startTime;
+  // Start time is null initially .
+  startTime: number | null;
```

Comment:
Why the start time is null initially? Can you share a link where we are assigning startTime to null? (Just wanted to verify that assigning null is expected!)

Reply:
@DubeySandeep `null` has been assigned in the constructor below, also here https://github.com/oppia/oppia/blob/78626a9925bf3f1f24b7a8eab882898818a8dbf4/core/templates/pages/exploration-player-page/services/stats-reporting.service.ts#L55

Reply:
that's not the startTime, right? Also, I'm not sure how the service is used if the startTime is set to null in the constructor? Do you see any use case of setting startTime to null in the constructor?

Reply:
@DubeySandeep The constructor is only used in the model itself at `static create` function, so each time we call `Stopwatch.create()` at some place the start time is initially set to null first, so we reset it after that with `.reset()` function and then get the current time.Below are two particular usecases:
https://github.com/oppia/oppia/blob/3e4444d54bf52e61cdd19afb8407f60b1ede8fdb/core/templates/services/playthrough.service.ts#L216
Also at `stats-reporting.service` first we initialize the stopwatch
https://github.com/oppia/oppia/blob/3e4444d54bf52e61cdd19afb8407f60b1ede8fdb/core/templates/pages/exploration-player-page/services/stats-reporting.service.ts#L147
and then reset it to get the current time
https://github.com/oppia/oppia/blob/3e4444d54bf52e61cdd19afb8407f60b1ede8fdb/core/templates/pages/exploration-player-page/services/stats-reporting.service.ts#L203